frontend/pulse3d/src/components/account/AdminAccountOptions.js

```diff
@@ -0,0 +1,174 @@
+import { useState, useEffect } from "react";
+import FormInput from "@/components/basicWidgets/FormInput";
+import ButtonWidget from "@/components/basicWidgets/ButtonWidget";
+import styled from "styled-components";
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  padding-top: 50px;
+  padding-right: 30px;
+  width: 100%;
+`;
+
+const SaveButtonTextContainer = styled.span`
+  font-style: italic;
+  font-size: 15px;
+  padding-right: 10px;
+  line-height: 3;
+`;
+
+const ErrorText = styled.span`
+  color: red;
+  font-style: italic;
+  text-align: left;
+  position: relative;
+  width: 150%;
+  font-size: 13px;
+`;
+
+const InputContainer = styled.div`
+  padding-right: 20%;
+  display: flex;
+  flex-direction: column;
+`;
+
+const UPDATE_STATUSES = {
+  NO_EDITS: "no_edits",
+  EDITING: "editing",
+  PENDING: "pending",
+  SUCCESS: "success",
+  FAILURE: "failure",
+};
+
+export default function AdminAccountOptions({ accountId }) {
+  const [accountSettings, setAccountSettings] = useState({});
+  const [accountSettingsEdits, setAccountSettingsEdits] = useState({});
+  const [errorMessages, setErrorMessages] = useState({});
+
+  const [updateStatus, setUpdateStatus] = useState(UPDATE_STATUSES.NO_EDITS);
+
+  const inProgress = updateStatus === UPDATE_STATUSES.PENDING;
+  const isButtonDisabled =
+    Object.values(errorMessages).some((val) => val.length > 0) ||
+    inProgress ||
+    // disable button if the current inputs are up to date
+    [UPDATE_STATUSES.NO_EDITS, UPDATE_STATUSES.SUCCESS].includes(updateStatus);
+
+  const getSaveButtonText = () => {
+    if (updateStatus === UPDATE_STATUSES.SUCCESS) {
+      return <SaveButtonTextContainer style={{ color: "green" }}>Update Successful!</SaveButtonTextContainer>;
+    } else if (updateStatus === UPDATE_STATUSES.FAILURE) {
+      return (
+        <SaveButtonTextContainer style={{ color: "red" }}>Error, Update Failed.</SaveButtonTextContainer>
+      );
+    }
+    return <></>;
+  };
+
+  // get account settings at load
+  useEffect(() => {
+    getAccountSettings();
+  }, []);
+
+  const getAccountSettings = async () => {
+    const res = await fetch(`${process.env.NEXT_PUBLIC_USERS_URL}/${accountId}`);
+    const data = await res.json();
+    setAccountSettings(data);
+    setAccountSettingsEdits(data);
+  };
+
+  const updateErrorMessages = (newErrorMessages) => {
+    setErrorMessages({
+      ...errorMessages,
+      ...newErrorMessages,
+    });
+  };
+
+  const updateAccountSettingsEdits = (newValues) => {
+    setUpdateStatus(UPDATE_STATUSES.EDITING);
+
+    const { alias: newAlias } = newValues;
+    if (newAlias != null) {
+      let errorMsg = "";
+      if (newAlias.length > 0) {
```

Comment:
What happens if the user inputs something and then backspaces to length of 0?

Reply:
submitting with the input set to an empty string is fine, the BE will interpret that as clearing the ID